sklearn/datasets/_lfw.py

```diff
@@ -109,7 +110,8 @@ def _check_fetch_lfw(data_home=None, funneled=True, download_if_missing=True):
         import tarfile
 
         logger.debug("Decompressing the data archive to %s", data_folder_path)
-        tarfile.open(archive_path, "r:gz").extractall(path=lfw_home)
+        with tarfile.open(archive_path, "r:gz") as fp:
+            tarfile_extractall(fp, path=lfw_home)
```

Comment:
This line being not covered was not covered before either in a PR (unless you triggered a scipy-dev build).

Any issues will be caught in a scheduled CI run on `main`